### PR TITLE
Working and playable in general (1.5.1)

### PR DESCRIPTION
**Pull request by ddos666TB**
texture names:
   -for items: unlocalizedname.png,
   -modules: itemModule(unlocalizedname for modules)/ClassName.png,
- remote orderer remoteOrdererItem(unlocalized name)/damagevalue.png
- logistic parts: like for orderer .../damagevalue.png

known issues:
+optifine gets crashed with forge 651

TODO:
fix crafting sign
bind 1 missing texure
apply patches from 1.4.7 repo

### CODE REVIEW

**Comment by GUIpsp** [comment]
Squish these commits and give the textures meaningful names please.

**Comment by AartBluestoke** [comment]
gejzer: thanks for these updates; i have little experience with the minecraft rendering system and hadn't had the time to get my head around it.

GUIpsp's comment about squashing would be to run "git rebase -i 66b09ed" and just replace pick with squash for non-major commits (see http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html). that keeps the changes, but doesn't add an extra message in the log.

(to be honest the entire codebase of commits over the last few months could do with a decent squash ...)

Also, feel free to come by #rs485 on irc.esper.net which is where most of the LP devs hang out.

**Comment by ddos666TB** [comment]
thx for feedback, i was using auto generated names and damage values for textures - for testing rendering engine, which texture did you mean?

**Comment by ddos666TB** [comment]
there are many hotfixes just for being able to compile (know about BC thingy)
copy paste isn't mine :)
about debug thing: there was/is token for ant build

**Comment by GUIpsp** [comment]
@gejzer well then, use this issue as a list of what needs changing :)

**Comment by ddos666TB** [comment]
yep thanks, but there are major things to do like rendering pipes :), it would be nice if you can help a bit

**Comment by GUIpsp** [comment]
@gejzer I might if I get around to it :)

**Comment by GUIpsp** [comment]
also, pretty sure package-simple already does what main_keepforge does

**Comment by ddos666TB** [comment]
ok keep going:D

File: common/logisticspipes/utils/SimpleInventory.java [hunk 1]
```diff
@@ -278,4 +278,16 @@ public boolean containsUndamagedItem(final ItemIdentifier item) {
...
+	@Override
+	public boolean isInvNameLocalized() {
+		// TODO Auto-generated method stub
+		return false;
+	}
+
+	@Override
+	public boolean isStackValidForSlot(int i, ItemStack itemstack) {
+		// TODO Auto-generated method stub
+		return false;
```
Comment (AartBluestoke) [inline common/logisticspipes/utils/SimpleInventory.java:291]: perhaps return true here -- the default should be "yes this item can go here"

File: common/logisticspipes/LogisticsPipes.java [hunk 2]
```diff
@@ -116,11 +119,12 @@
 import cpw.mods.fml.common.registry.TickRegistry;
 import cpw.mods.fml.relauncher.FMLInjectionData;
 import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
 
 @Mod(
 		modid = "LogisticsPipes|Main",
 		name = "Logistics Pipes",
-		version = "%VERSION%",
+		version = "${lp.version.full}",
```
Comment (davboecki) [inline common/logisticspipes/LogisticsPipes.java]: Please revert this change

File: common/logisticspipes/LogisticsPipes.java [hunk 3]
```diff
@@ -150,8 +154,8 @@
 	//Log Requests
 	public static boolean DisplayRequests;
 
-	public static boolean DEBUG = "%DEBUG%".equals("%" + "DEBUG" + "%") || "%DEBUG%".equals("true");
-	public static String MCVersion = "%MCVERSION%";
+	public static boolean DEBUG = "%DEBUG%".equals("%" + "DEBUG" + "%") || "${DEBUG}".equals("true");
+	public static String MCVersion = "1.5.1";
```
Comment (davboecki) [inline common/logisticspipes/LogisticsPipes.java]: And these two.

File: .travis.yml [hunk 4]
```diff
@@ -12,4 +12,4 @@ before_script:
   - mv .gitignore src/
   - mv build.xml src/
   - cd src
-script: ant test
+script: ant main
```
Comment (davboecki) [inline .travis.yml]: Travis is only ment to test if builds can work for pull requests not for the real building cause it can't sign the mod.

File: build.xml [hunk 5]
```diff
@@ -16,27 +15,30 @@
...
+	<property name="ic2.dir"              value="${mcp.dir}/unpack/ic2"/>
+	<property name="cc.dir"               value="${mcp.dir}/unpack/cc"/>
 
 	<property name="clientsrc.dir"        value="${mcp.dir}/src/minecraft"/>
 	<property name="commonsrc.dir"        value="${clientsrc.dir}"/>
-
-	<property name="downloadserver"       value="http://www.theZorro266.com"/>
-	<property name="downloadserver.dir"   value="rs485/"/>
+	
+	<property file="${clientsrc.dir}/fmlversion.properties" />
```
Comment (davboecki) [inline build.xml:26]: How is this supposed to work on an fresh install without fml installed?

Comment (ddos666TB) [inline build.xml:26]: these property will be unset, ant detect it via ant-contrib if isset

File: .travis.yml (hunk 4 above)
Comment (AartBluestoke) [inline .travis.yml]: test builds different to main -- the debging flag. shouldn't travis build main, as we have had the occasional debug flag bug, which caused a build fail?

Comment (davboecki) [inline .travis.yml]: test is ment for travis. If there is a building bug the building script should be fixed. main is for compiling at home if i remember correctly, ci and ci-dev for jenkins and test for travis. if you want travis todo something else than change the test task.

**Comment by davboecki** [comment]
Could you also clean up the unneeded api parts you added.

**Comment by ddos666TB** [comment]
ok i'll

**Comment by AartBluestoke** [comment]
thanks for this :)
Edit: 1.5 code added to jenkins.
